### DISCUSSION TITLE
How do you use the option `jovian.steam.desktopSession`

### DISCUSSION THREAD

**Discussion by dnkmmr69420** [Jovian-Experiments/Jovian-NixOS#179] [Q&A] [answered]
Like can someone give me an example of having the desktop session being KDE wayland?

`jovian.steam.desktopSession`

**Comment by samueldr** [comment, marked as answer]
The session name is now `"plasma"`.

It was previously `"plasmawayland"`.

**Reply by Azelphur** [reply]
for future googlers that land here as I did, this has now changed to "plasma" 

```
       error: Desktop session 'plasmawayland' not found.
       Valid values for 'jovian.steam.desktopSession' are:
         gamescope-wayland
         plasma
         plasmax11
```

**Comment by dnkmmr69420** [comment]
oh ok so like this?

```nix
jovian.steam.desktopSession = plasmawayland;
```

**Comment by dnkmmr69420** [comment]
or like this?

```nix
jovian.steam.desktopSession = "plasmawayland";
```

**Comment by samueldr** [comment]
You could try either, and see, but it's the latter, since otherwise it would refer to the variable name `plasmawayland`, which wouldn't be a thing in scope.

**Comment by IceDBorn** [comment]
Does this work when autologin is not enabled? Also, I tried enabling autologin and set `jovian.steam.desktopSession` to `Hyprland` and `hyprland`, but it just boots back to the steam deck UI.

**Comment by b-m-f** [comment]
> Does this work when autologin is not enabled? Also, I tried enabling autologin and set `jovian.steam.desktopSession` to `Hyprland` and `hyprland`, but it just boots back to the steam deck UI.

I experience the same when setting desktopSession to `gnome` with:
```nix
{ ... }: {

  services.xserver.enable = true;
  services.xserver.desktopManager.gnome.enable = true;

  jovian.steam = {
    enable = true;
    autoStart = true;
    user = "user";
    desktopSession = "gnome";
  };
}

```

Edit:

```nix
{ ... }: {

  services.xserver.enable = true;
  services.xserver.desktopManager.plasma5.enable = true;

  jovian.steam = {
    enable = true;
    autoStart = true;
    user = "user";
    desktopSession = "plasma";
  };
}

```
Works fine. So it should not be anything else in my config.

**Comment by xiro-codes** [comment]
I just gonna mention the session name can be found by running `nix build nixpkgs#<desktop package name>` and browsing the result/share/wayland-sessions/ folder its the the name of the file in there

**Reply by imadnyc** [reply]
Does anyone know how to find the gnome session name? I built gnome.gnome-shell and gnome.gnome-session, and only the session showed anything useful, to put the session as GNOME instead of gnome. However, neither worked. It's confusing, since I see configs on github that just use gnome or gnome-wayland, neither of which work for me.

**Reply by DylanRJohnston-FZ** [reply]
Hey @imadnyc I was having this exact same issue. I assume you were following the config in [here](https://github.com/acristoffers/dotfiles/blob/007cb53614db4ff25921f504214e6eb401130d76/dotfiles/home-manager/nixos/steamdeck.nix) which is the only example I could find of `gnome-wayland`. 

You'll see in that config that they disable gdm when auto start is enabled

https://github.com/acristoffers/dotfiles/blob/007cb53614db4ff25921f504214e6eb401130d76/dotfiles/home-manager/nixos/steamdeck.nix#L22

But the hidden part is the config inherits from base where gnome is enabled https://github.com/acristoffers/dotfiles/blob/007cb53614db4ff25921f504214e6eb401130d76/dotfiles/home-manager/nixos/base.nix#L77 and pulse audio is disabled https://github.com/acristoffers/dotfiles/blob/007cb53614db4ff25921f504214e6eb401130d76/dotfiles/home-manager/nixos/base.nix#L101.

I needed to add those to my config to make it work.

```nix
  services.xserver.desktopManager.gnome.enable = true;
  hardware.pulseaudio.enable = lib.mkForce false;
```

This is my full config

```nix
{ common, pkgs, modules, lib, ... }: {
  imports = with common; [
    user
    modules.jovian
    wifi
    zsh
    ./hardware-configuration.nix
  ];

  boot.loader.systemd-boot.enable = true;
  boot.loader.efi.canTouchEfiVariables = true;

  networking.networkmanager.enable = true;
  services.openssh.enable = true;
  services.openssh.ports = [ 2022 ];

  services.xserver.desktopManager.gnome.enable = true;
  hardware.pulseaudio.enable = lib.mkForce false;

  jovian = {
    steam.enable = true;
    steam.autoStart = true;
    steam.user = "dylanj";
    steam.desktopSession = "gnome-wayland";
    devices.steamdeck.enable = true;
  };

  environment.systemPackages = with pkgs; [
    gnomeExtensions.dash-to-dock
    # jupiter-docker-updater-bin
    steamdeck-firmware
    git
    vim
    steam
  ];

  system.stateVersion = "23.11";
}
```

**Reply by imadnyc** [reply]
Thanks for the writeup. Unfortunately, I did both already, and don't know why it doesn't work for me. Below, I've attached my config.

```
{ inputs, outputs, lib, config, pkgs, ... }: {
  imports = [
    inputs.jovian.nixosModules.jovian
    inputs.home-manager.nixosModules.home-manager 
    ./hardware-configuration.nix
    ../../generic-pc/system/default.nix
 ];

  jovian = {
    devices.steamdeck = {
      enable = true;
    };
    steam = {
      enable = true;
      autoStart =true;     
      user = "dre";
      desktopSession = "gnome-wayland";
    };
    # decky-loader = {
    #   enable = true;
    # };
  };
  services = {

    xserver.enable = true;
    xserver.displayManager.gdm.enable = lib.mkForce false;
    # xserver.desktopManager.gnome.enable = lib.mkForce false; #it inherits it being true from generic PC - also doesn't work when explicitly to true
    pipewire.wireplumber.enable = true;
    printing.enable = true;
    udev.packages = [pkgs.gnome.gnome-settings-daemon];
    power-profiles-daemon.enable = true;
  };

  hardware = {
    pulseaudio.enable = false;
    enableRedistributableFirmware = true;
    steam-hardware.enable = true;
  };
}
```
I've removed unnecessary parts of the config, but I'm pretty sure that this should work. I can't figure out why this doesn't work - it looks the same as any other config. Like the commentor above, this works flawlessly with plasma5, so maybe it's something to do with the session name? Though you just said that `gnome-wayland` works. Maybe it's something that it's inheriting, so I'm going to try and see if making a config without imports helps, but I really don't think so. 

I just checked, and there's nothing from there that can interfere, since everything is manually overriden in the deck config. Also, gnome definitely is installed, since i can launch the wayland session from the command line.

**Reply by imadnyc** [reply]
I've also taken your config verbatim and it still doesn't launch. Haven't got a clue.

**Reply by imadnyc** [reply]
Sorry to keep adding - using this as a sort of dev log so other people aren't as confused as me. 

When loading my config in my nix repl via `:lf <path to flake>`, I noticed that the sessionNames that it accepts are only

```
nix-repl> nixosConfigurations.nix-deck.config.services.xserver.displayManager.sessionData.sessionNames
[ "gamescope-wayland" "gnome" "gnome-xorg" ]

```

This makes me wonder, how does gnome-wayland work in your config, if that's not a sessionName?

**Reply by DylanRJohnston-FZ** [reply]
Here's the repo for my config if you want to do any kind of deeper comparison https://github.com/DylanRJohnston/nixos/blob/main/hosts/steamdeck/configuration.nix

**Comment by minego** [comment]
I'm running into the same issue... I'm trying to use dwl, but it should be very similar to hyperland. They are both wlroots based.

I checked and it is listed in the session names ```
nix-repl> nixosConfigurations.nix-deck.config.services.xserver.displayManager.sessionData.sessionNames
[ "gamescope-wayland" "dwl" ]
```

And, I'm using the same config on other machines with greetd, and that is able to login to dwl properly, but quitting to desktop just restarted the steam deck UI.

Does anyone have tips for debugging this?

**Comment by minego** [comment]
I managed to get mine working finally! This was very confusing, but hopefully this can help some of you...

Originally I was specifying my session:
```
services.xserver.displayManager.session = [{
			manage									= "desktop";
			name									= "dwl";
			start = ''
                ${pkgs.dwl}/bin/dwl &
                waitPID=$!
                '';
		}];
```

This works on my desktop systems using greetd with regreet, but does NOT work with Jovian's greeter. So, I spent some time digging through source and trying to understand it. So, I looked at a compositor that we know does work:

https://github.com/Jovian-Experiments/Jovian-NixOS/blob/e2c026d8efea340d2a2dcc56775212979dd51ef2/pkgs/gamescope-session/default.nix#L151

That line was something I had not seen before. So, I updated my dwl package to provide that:
```
passthru.providedSessions = [ "dwl" ];
```
This syntax assumes that the package provides a session .desktop file, and my package does. If yours doesn't then you'll also need to update it to create that file in `$out/share/wayland-sessions/$name.desktop`

Then in my configuration I replaced the section above with:
```
		services.xserver.displayManager.sessionPackages = [
			pkgs.dwl-unwrapped
		];
```

So, with that it is all working!
